File: python/questions/groupAnagrams.py
from typing import List

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dt = dict()
        for s in strs:
            k = "".join(sorted(s))
            if k in dt:
                dt[k].append(s)
            else:
                dt[k] = [s]
        return [sl for _, sl in dt.items()]

File: typescript/questions/groupAnagrams.ts
function groupAnagrams(strs: string[]): string[][] {
  const mp: Map<string, string[]> = new Map();
  for (const s of strs) {
    const sv = s.split("").sort().join("");
    if (!mp.has(sv)) {
      mp.set(sv, []);
    }
    mp.get(sv)?.push(s);
  }
  return Array.from(mp.values());
}
